api/api.py

```diff
@@ -100,33 +101,50 @@ def host_response():
         action = str(request.json['action'])
         target = request.json['target']
         if action == 'blacklist':
-            utils.flush_ipset_list('blacklist-user')
-            for ip in target:
-                with lock:
-                    if (len(ip) >= 7) and (ip not in homenet.user_blacklist) and (ip not in homenet.user_whitelist):
-                        homenet.user_blacklist.append(ip)
-                        utils.add_ip_ipset_blacklist(ip, 'blacklist-user')
-                        log.debug('FG-INFO: IP ' + ip + 'added to user blacklist')
+            if re.search('[a-zA-Z]', target):
+                for domain in target:
+                    with lock:
+                        if (domain not in homenet.user_whitelist) and (domain not in homenet.user_blacklist):
+                            homenet.user_blacklist.append(domain)
```

Comment:
The object "user_blacklist" was used to store only IP addresses. Adding domains instead of IPs to this list might cause undesired issues somewhere else in the program (e.g. while adding the items from this list to ipset). I suggest you to create and use a dedicated list instead in the Network class located in objects.py.